irohad/maintenance/metrics.cpp

```diff
@@ -4,93 +4,139 @@
-    listen_addr_port = listen_addr;
+    listen_addr_port_ = listen_addr;
   } else if(std::regex_match(listen_addr,port_matcher)) {
-    listen_addr_port = "127.0.0.1";
+    listen_addr_port_ = "127.0.0.1";
     if (listen_addr[0] != ':')
-      listen_addr_port += ":";
-    listen_addr_port += listen_addr;
+      listen_addr_port_ += ":";
+    listen_addr_port_ += listen_addr;
   } else {
-    return nullptr;
+    return;
   }
 
-  // create a metrics registry
   // @note it's the users responsibility to keep the object alive
-  auto registry = std::make_shared<Registry>();
-
-  // Just for example
-  std::thread([registry,listen_addr_port](){
-      // create an http server running on addr:port
-      Exposer exposer{listen_addr_port};
-
-      // ask the exposer to scrape the registry on incoming HTTP requests
-      exposer.RegisterCollectable(registry);
-
-    // add a new counter family to the registry (families combine values with the
-    // same name, but distinct label dimensions)
-    //
-    // @note please follow the metric-naming best-practices:
-    // https://prometheus.io/docs/practices/naming/
-    auto& packet_counter = BuildCounter()
-        .Name("observed_packets_total")
-        .Help("Number of observed packets")
-        .Register(*registry);
+  registry_ = std::make_shared<Registry>();
 
-    // add and remember dimensional data, incrementing those is very cheap
-    auto& tcp_rx_counter =
-        packet_counter.Add({{"protocol", "tcp"}, {"direction", "rx"}});
-    auto& tcp_tx_counter =
-        packet_counter.Add({{"protocol", "tcp"}, {"direction", "tx"}});
-    auto& udp_rx_counter =
-        packet_counter.Add({{"protocol", "udp"}, {"direction", "rx"}});
-    auto& udp_tx_counter =
-        packet_counter.Add({{"protocol", "udp"}, {"direction", "tx"}});
+  // create an http server running on addr:port
+  exposer_ = std::make_shared<Exposer>(listen_addr_port_);
 
-    // add a counter whose dimensional data is not known at compile time
-    // nevertheless dimensional values should only occur in low cardinality:
-    // https://prometheus.io/docs/practices/naming/#labels
-    auto& http_requests_counter = BuildCounter()
-        .Name("http_requests_total")
-        .Help("Number of HTTP requests")
-        .Register(*registry);
+  // ask the exposer_ to scrape the registry_ on incoming HTTP requests
+  exposer_->RegisterCollectable(registry_);
 
-    for (;;) {
-      std::this_thread::sleep_for(std::chrono::seconds(1));
-      const auto random_value = std::rand();
+  auto&block_height_gauge = BuildGauge()
+                                .Name("blocks_height")
+                                .Help("Total number of blocks in chain")
+                                //.Labels({{"label","a_metter"}})
+                                .Register(*registry_);
+  auto&block_height = block_height_gauge.Add({});//{{"value", "some"}});
+  block_height.Set(storage_->getBlockQuery()->getTopBlockHeight());
```

Comment:
What happens while reindexing WSV is in progress?

Reply:
Never mind. Will look at this moment. Anyway this is a early start up. So i think possibiblity of race condition is lowest

Reply:
Merics run exatly after irohad initialized and after WSV rebuilt.